### ISSUE TITLE
Nested Docker containers on Windows?

### ISSUE THREAD

**Issue by konradzuse** [coder/coder#10632]
I'm struggling to get docker working inside a workspace running on Docker Windows.

Sysbox only works on Linux. Is there a way to get this scenario working on Windows?

**Comment by matifali** [comment]
@konradzuse, could you share your template and use case on why you want to use containers inside Windows containers?

**Comment by konradzuse** [comment]
I've just used the starter docker container. We have a legacy Laravel project that consists of a number of docker containers; Laravel backend, mySQL database container, React frontend container, Nginx etc

**Comment by matifali** [comment]
@konradzuse the starter docker template is based on the Ubuntu Docker image. Your issue says that you are using Docker Windows. Can you please explain your setup on where Coder is installed and your Docker setup?

**Comment by konradzuse** [comment]
Sorry if I wasn't clear. I've installed Coder on a Windows 11 host, then within the coder workspace I wish to use Docker.

**Comment by matifali** [comment]
@konradzuse is Docker installed on the same Windows machine?
is Docker using the WSL backend or the Windows Containers backend?

**Comment by konradzuse** [comment]
Yes, Docker is installed on the Windows host, it's currently using hyper-V

**Comment by matifali** [comment]
There are a few things to note here,
1. You can not use the starter Docker template with Windows Containers as it is a Ubuntu template. (We currently do not publish a Windows docker template)
2. Sysbox is designed to be run in Linux containers, so you need a Linux Docker host. You can also Docker Desktop with WSL2 backend for non-docker-in-docker workspaces, but this also [does not support sysbox](https://github.com/nestybox/sysbox/issues/32).

**Comment by konradzuse** [comment]
That's what I thought. Thank you for confirming.

**Comment by matifali** [comment]
Also, if you need Windows containers, you should probably start here: https://learn.microsoft.com/en-us/virtualization/windowscontainers/quick-start/set-up-environment?tabs=dockerce.

**Comment by konradzuse** [comment]
Actually, the workspace itself should be Linux. It's only the physical host that it runs on is Windows running Docker desktop.

**Comment by matifali** [comment]
@konradzuse Then you should use the WSL2 backend.
